src/test/enhetstester/kotlin/no/nav/familie/ks/sak/kjerne/beregning/BeregningServiceTest.kt

```diff
@@ -358,4 +359,168 @@ class BeregningServiceTest {
 
         assertEquals(0, tilkjentYtelseIverksattMotØkonomi.size)
     }
+
+    @Nested
+    inner class OppdaterTilkjentYtelsePåBehandlingTest {
+        @Test
+        fun `skal oppdatere tilkjentytelse basert på endretutbetalings andeler uavhengig av overlapp med andeler dersom behandlingen er automatisk`() {
+            // Arrange
+            val behandling = lagBehandling(opprettetÅrsak = BehandlingÅrsak.LOVENDRING_2024)
+            val personopplysningGrunnlag = lagPersonopplysningGrunnlag(behandling.id)
+            val vilkårsvurdering = lagVilkårsvurdering(behandling = behandling)
+
+            val endretUtbetalingMedAndel1 = EndretUtbetalingAndelMedAndelerTilkjentYtelse(lagEndretUtbetalingAndel(id = 1), emptyList())
+            val endretUtbetalingMedAndel2 = EndretUtbetalingAndelMedAndelerTilkjentYtelse(lagEndretUtbetalingAndel(id = 2), emptyList())
+
+            val endretUtbetalingAndelerMedAndeler =
+                listOf(
+                    endretUtbetalingMedAndel1,
+                    endretUtbetalingMedAndel2,
+                )
+
+            val lagretEndretUtbetalingAndelerMedAndelerSlot = slot<MutableList<EndretUtbetalingAndelMedAndelerTilkjentYtelse>>()
+            val opprettetTilkjentYtelse = mockk<TilkjentYtelse>()
+
+            every { mockAndelerTilkjentYtelseOgEndreteUtbetalingerService.finnEndreteUtbetalingerMedAndelerTilkjentYtelse(behandling.id) } returns endretUtbetalingAndelerMedAndeler
+            every { mockTilkjentYtelseRepository.slettTilkjentYtelseForBehandling(behandling) } just runs
+            every { mockTilkjentYtelseService.beregnTilkjentYtelse(vilkårsvurdering, personopplysningGrunnlag, capture(lagretEndretUtbetalingAndelerMedAndelerSlot)) } returns opprettetTilkjentYtelse
+            every { mockTilkjentYtelseRepository.save(opprettetTilkjentYtelse) } returns opprettetTilkjentYtelse
+
+            // Act
+            beregningService.oppdaterTilkjentYtelsePåBehandling(behandling, personopplysningGrunnlag, vilkårsvurdering)
+
+            // Assert
+            verify(exactly = 1) { mockAndelerTilkjentYtelseOgEndreteUtbetalingerService.finnEndreteUtbetalingerMedAndelerTilkjentYtelse(behandling.id) }
+            verify(exactly = 1) { mockTilkjentYtelseRepository.slettTilkjentYtelseForBehandling(behandling) }
+            verify(exactly = 1) { mockTilkjentYtelseService.beregnTilkjentYtelse(vilkårsvurdering, personopplysningGrunnlag, capture(lagretEndretUtbetalingAndelerMedAndelerSlot)) }
+            verify(exactly = 1) { mockTilkjentYtelseRepository.save(opprettetTilkjentYtelse) }
+
+            val lagretEndretUtbetalingAndelerMedAndeler = lagretEndretUtbetalingAndelerMedAndelerSlot.captured
+
+            assertThat(lagretEndretUtbetalingAndelerMedAndeler.size).isEqualTo(2)
```

Comment:
Småpirk men man vurdere:

```suggestion
            assertThat(lagretEndretUtbetalingAndelerMedAndeler´).hasSize(2)
```